IDEA import fails with `buildSrc` project and `org.gradle.java.home` specified

Given a simple project with:
- a `buildSrc` project
- a top-level `gradle.properties` setting `org.gradle.java.home` to a JDK 9+

When attempting to import/sync in Intellij IDEA, the daemon process cannot be started.

### Expected Behavior
Build will sync/import without error.

### Current Behavior
Sync fails after initial "CONFIGURE SUCCESSFUL:
```
> Task :buildSrc:compileJava UP-TO-DATE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:assemble UP-TO-DATE
> Task :buildSrc:compileTestJava NO-SOURCE
> Task :buildSrc:compileTestGroovy NO-SOURCE
> Task :buildSrc:processTestResources NO-SOURCE
> Task :buildSrc:testClasses UP-TO-DATE
> Task :buildSrc:test NO-SOURCE
> Task :buildSrc:check UP-TO-DATE
> Task :buildSrc:build UP-TO-DATE

CONFIGURE SUCCESSFUL in 0s
Unable to start the daemon process.
This problem might be caused by incorrect configuration of the daemon.
For example, an unrecognized jvm option is used.
Please refer to the user guide chapter on the daemon at https://docs.gradle.org/5.1.1/userguide/gradle_daemon.html
Please read the following process output to find out more:
-----------------------
Error: Could not find or load main class java.base.java.lang=ALL-UNNAMED
Caused by: java.lang.ClassNotFoundException: java.base.java.lang=ALL-UNNAMED

```

![image](https://user-images.githubusercontent.com/179734/51857396-5fb90580-22ef-11e9-818b-919612ae0509.png)

### Context
This appears to be an issue when IDEA attempts to separately get the model for the `buildSrc` build.

Symptoms are similar to #7045 but I suspect the underlying cause is different. This is an issue starting the Gradle daemon itself, whereas #7045 is related to starting a Groovy compiler daemon.

Possibly related issues:
https://youtrack.jetbrains.com/issue/IDEA-202975
https://youtrack.jetbrains.com/issue/IDEA-203429

### Steps to Reproduce (for bugs)

Clone https://github.com/baron1405/gradle7045 and follow instructions in the README.

Thanks to @baron1405 for this reproducer.

--- a/subprojects/launcher/src/integTest/groovy/org/gradle/launcher/daemon/DaemonFeedbackIntegrationSpec.groovy
+++ b/subprojects/launcher/src/integTest/groovy/org/gradle/launcher/daemon/DaemonFeedbackIntegrationSpec.groovy
@@ -69,7 +69,9 @@ task sleep {
         then:
         def ex = thrown(Exception)
         ex.message.contains(DaemonMessages.UNABLE_TO_START_DAEMON)
-        ex.message.contains("-Xyz")
+        ex.message.contains("Process command line:")
+        ex.message.contains("-Dorg.gradle.jvmargs=-Xyz")
+        ex.message.contains("Unrecognized option: -Xyz")
     }
 
     def "daemon log contains all necessary logging"() {

--- a/subprojects/launcher/src/main/java/org/gradle/launcher/daemon/client/DaemonGreeter.java
+++ b/subprojects/launcher/src/main/java/org/gradle/launcher/daemon/client/DaemonGreeter.java
@@ -16,11 +16,14 @@
 
 package org.gradle.launcher.daemon.client;
 
+import com.google.common.base.Joiner;
 import org.gradle.api.GradleException;
 import org.gradle.api.internal.DocumentationRegistry;
 import org.gradle.launcher.daemon.bootstrap.DaemonStartupCommunication;
 import org.gradle.launcher.daemon.diagnostics.DaemonStartupInfo;
 import org.gradle.launcher.daemon.logging.DaemonMessages;
+
+import java.util.List;
 
 public class DaemonGreeter {
     private final DocumentationRegistry documentationRegistry;
@@ -29,10 +32,10 @@ public class DaemonGreeter {
         this.documentationRegistry = documentationRegistry;
     }
 
-    public DaemonStartupInfo parseDaemonOutput(String output) {
+    public DaemonStartupInfo parseDaemonOutput(String output, List<String> startupArgs) {
         DaemonStartupCommunication startupCommunication = new DaemonStartupCommunication();
         if (!startupCommunication.containsGreeting(output)) {
-            throw new GradleException(prepareMessage(output));
+            throw new GradleException(prepareMessage(output, startupArgs));
         }
         String[] lines = output.split("\n");
         //Assuming that the diagnostics were printed out to the last line. It's not bullet-proof but seems to be doing fine.
@@ -40,13 +43,14 @@ public class DaemonGreeter {
         return startupCommunication.readDiagnostics(lastLine);
     }
 
-    private String prepareMessage(String output) {
+    private String prepareMessage(String output, List<String> startupArgs) {
         StringBuilder sb = new StringBuilder();
         sb.append(DaemonMessages.UNABLE_TO_START_DAEMON);
         sb.append("\nThis problem might be caused by incorrect configuration of the daemon.");
         sb.append("\nFor example, an unrecognized jvm option is used.");
         sb.append("\nPlease refer to the User Manual chapter on the daemon at ");
         sb.append(documentationRegistry.getDocumentationFor("gradle_daemon"));
+        sb.append("\nProcess command line: ").append(Joiner.on(" ").join(startupArgs));
         sb.append("\nPlease read the following process output to find out more:");
         sb.append("\n-----------------------\n");
         sb.append(output);

--- a/subprojects/launcher/src/main/java/org/gradle/launcher/daemon/client/DefaultDaemonStarter.java
+++ b/subprojects/launcher/src/main/java/org/gradle/launcher/daemon/client/DefaultDaemonStarter.java
@@ -175,7 +175,7 @@ public class DefaultDaemonStarter implements DaemonStarter {
                 execActionFactory.stop();
             }
 
-            return daemonGreeter.parseDaemonOutput(outputConsumer.getProcessOutput());
+            return daemonGreeter.parseDaemonOutput(outputConsumer.getProcessOutput(), args);
         } catch (GradleException e) {
             throw e;
         } catch (Exception e) {

--- a/subprojects/launcher/src/test/groovy/org/gradle/launcher/daemon/bootstrap/DaemonGreeterTest.groovy
+++ b/subprojects/launcher/src/test/groovy/org/gradle/launcher/daemon/bootstrap/DaemonGreeterTest.groovy
@@ -27,6 +27,7 @@ import spock.lang.Specification
 class DaemonGreeterTest extends Specification {
 
     def registry = Mock(DocumentationRegistry)
+    def args = ["foo", "bar"]
 
     def "parses the process output"() {
         given:
@@ -42,7 +43,7 @@ another line of output...
         def output = new String(outputStream.toByteArray())
 
         when:
-        def daemonStartupInfo = new DaemonGreeter(registry).parseDaemonOutput(output)
+        def daemonStartupInfo = new DaemonGreeter(registry).parseDaemonOutput(output, args)
 
         then:
         daemonStartupInfo.address == address
@@ -60,17 +61,18 @@ another line of output..."""
         ExecResult result = Mock()
 
         when:
-        new DaemonGreeter(registry).parseDaemonOutput(output);
+        new DaemonGreeter(registry).parseDaemonOutput(output, args);
 
         then:
         def ex = thrown(GradleException)
         ex.message.contains(DaemonMessages.UNABLE_TO_START_DAEMON)
+        ex.message.concat("Process command line: [foo, bar]")
         ex.message.contains("hey joe!")
     }
 
     def "shouts if daemon broke completely..."() {
         when:
-        new DaemonGreeter(registry).parseDaemonOutput("")
+        new DaemonGreeter(registry).parseDaemonOutput("", args)
 
         then:
         def ex = thrown(GradleException)

